internal/config/config.go

```diff
@@ -1,144 +1,126 @@
-	Scheme string
-	// If set, Port will discard all endpoints discovered that do not use this specified port. Otherwise, all endpoints will be considered.
-	Port int
-	// PodLabel is the selector used to filter Endpoints.
-	PodLabel string
-	// Namespace can be used to restric the search to a particular namespace.
-	Namespace string
-	// If set, Distributed will instruct the integration to scrape all KSM endpoints rather than just the first one.
-	Distributed bool
-
-	Enabled bool
+	StaticURL   string `mapstructure:"staticURL"`
+	Scheme      string `mapstructure:"scheme"`
+	Port        int    `mapstructure:"port"`
+	Selector    string `mapstructure:"selector"`
+	Namespace   string `mapstructure:"namespace"`
+	Distributed bool   `mapstructure:"distributed"`
+	Enabled     bool   `mapstructure:"enabled"`
+}
+
+type Kubelet struct {
+	Enabled bool   `mapstructure:"enabled"`
+	Port    int32  `mapstructure:"port"`
+	Scheme  string `mapstructure:"scheme"`
 }
 
 type ControlPlane struct {
-	ETCD              ETCD
-	APIServer         APIServer
-	Scheduler         Scheduler
-	ControllerManager ControllerManager
-	Enabled           bool
+	Enabled           bool              `mapstructure:"enabled"`
+	ETCD              ETCD              `mapstructure:"etcd"`
+	APIServer         APIServer         `mapstructure:"apiServer"`
+	ControllerManager ControllerManager `mapstructure:"controllerManager"`
+	Scheduler         Scheduler         `mapstructure:"scheduler"`
+}
+
+type ETCD struct {
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
 type APIServer struct {
-	APIServerEndpointURL string
-	APIServerSecurePort  string
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
-type ETCD struct {
-	EtcdEndpointURL        string
-	EtcdTLSSecretNamespace string
-	EtcdTLSSecretName      string
+type ControllerManager struct {
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
 type Scheduler struct {
-	SchedulerEndpointURL string
+	Enabled        bool                       `mapstructure:"enabled"`
+	StaticEndpoint StaticEndpoint             `mapstructure:"staticEndpoint"`
+	Autodiscover   []AutodiscoverControlPlane `mapstructure:"autodiscover"`
 }
 
-type ControllerManager struct {
-	ControllerManagerEndpointURL string
+type AutodiscoverControlPlane struct {
+	Namespace string `mapstructure:"namespace"`
+	Selector  string `mapstructure:"selector"`
+	MatchNode bool   `mapstructure:"matchNode"`
+	URL       string `mapstructure:"url"`
```

Comment:
I think this will need to be `[]string` to support different common options. [currently](https://github.com/newrelic/nri-kubernetes/blob/main/src/controlplane/components.go#L186) API server has 2 preconfigured. But perhaps @roobre  or @kang-makes has other opinions.

Reply:
I can refactor this in my pr if we decided to change it